apps/worker/src/app/workflow/services/metric-queue-active.service.ts

```diff
@@ -0,0 +1,115 @@
+import { WorkerOptions } from 'bullmq';
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { QueueService } from '@novu/application-generic';
+import { checkinForCronJob } from '../../shared/utils';
+import * as process from 'process';
+import { JobTopicNameEnum } from '@novu/shared';
+
+const LOG_CONTEXT = 'MetricQueueService';
+const METRIC_JOB_ID = 'metric-job';
+
+@Injectable()
+export class MetricQueueActiveService extends QueueService<Record<string, never>> {
+  constructor(@Inject('BULLMQ_LIST') private token_list: QueueService[]) {
+    super(JobTopicNameEnum.METRICS_ACTIVE);
+
+    this.bullMqService.createWorker(this.name, this.getWorkerProcessor(), this.getWorkerOpts());
+
+    this.bullMqService.worker.on('completed', async (job) => {
+      await checkinForCronJob(process.env.ACTIVE_CRON_ID);
+      Logger.verbose('Metric job Completed', job.id, LOG_CONTEXT);
+    });
+
+    this.bullMqService.worker.on('failed', async (job, error) => {
+      Logger.verbose('Metric job failed', LOG_CONTEXT, error);
+    });
+
+    this.addToQueueIfMetricJobExists();
+  }
```

Comment:
```suggestion
export class MetricQueueActiveService extends QueueService<Record<string, never>> {
  constructor(@Inject('BULLMQ_LIST') private token_list: QueueService[]) {
    super(JobTopicNameEnum.METRICS_ACTIVE);

    if (process.env.NOVU_MANAGED_SERVICE === 'true' && process.env.NEW_RELIC_LICENSE_KEY) {
      this.bullMqService.createWorker(this.name, this.getWorkerProcessor(), this.getWorkerOpts());

      this.bullMqService.worker.on('completed', async (job) => {
        await checkinForCronJob(process.env.ACTIVE_CRON_ID);
        Logger.verbose('Metric job Completed', job.id, LOG_CONTEXT);
      });

      this.bullMqService.worker.on('failed', async (job, error) => {
        Logger.verbose('Metric job failed', LOG_CONTEXT, error);
      });

      this.addToQueueIfMetricJobExists();
    }
  }
```
This is my suggestion for this comment: https://github.com/novuhq/novu/pull/4060/files#r1311743616

Thus for self hosted users we do not initialise the worker as they are not going to use it or need it. We might think in the future how to redo this if users request a feature to have a similar uptime implementation like us.